packages/web-wallet/pages/contacts/show/index.tsx

```diff
@@ -0,0 +1,550 @@
+import React, {FC, ReactElement, useEffect, useState} from 'react'
+import {useParams} from 'react-router-dom'
+import {HttpError, useOne, useTranslation} from '@refinedev/core'
+import {CredentialRole, IBasicCredentialLocaleBranding, Party} from '@sphereon/ssi-sdk.data-store'
+import {OpenID4VCIClient} from '@sphereon/oid4vci-client'
+import {CredentialStatus, TabViewRoute} from '@sphereon/ui-components.core'
+import {ContactViewItem, SSITabView} from '@sphereon/ui-components.ssi-react'
+import {credentialLocaleBrandingFrom} from '@sphereon/ssi-sdk.oid4vci-holder/dist/agent/OIDC4VCIBrandingMapper'
+import PageHeaderBar from '@components/bars/PageHeaderBar'
+import {staticPropsWithSST} from '@/src/i18n/server'
+import {
+  CredentialConfigurationSupported,
+  CredentialConfigurationSupportedV1_0_13,
+  CredentialsSupportedDisplay
+} from '@sphereon/oid4vci-common'
+import agent from "@agent";
+import CredentialCatalogView from "@components/views/CredentialCatalogView";
+import style from './index.module.css'
+import {CredentialCatalogItem} from '@typings';
+
+enum ContactDetailsTabRoute {
+  INFO = 'info',
+  ACTIVITY = 'activity',
+  CREDENTIAL_CATALOG = 'credentialCatalog',
+  RELATIONS = 'relations',
+  IDENTIFIERS = 'identifiers',
+}
+
+type getCredentialBrandingArgs = {
+  credentialsSupported: Record<string, CredentialConfigurationSupported>
+}
+
+type SelectCredentialLocaleBrandingArgs = {
+  locale?: string
+  localeBranding?: Array<IBasicCredentialLocaleBranding>
+}
+
+const ShowContactDetails: FC = (): ReactElement => {
+  const {translate, getLocale} = useTranslation()
+  const params = useParams()
+  const {id} = params
+  const [credentialsSupported, setCredentialsSupported] = useState<Record<string, CredentialConfigurationSupportedV1_0_13> | undefined>(undefined)
+  const [credentialCatalogItems, setCredentialCatalogItems] = useState<Array<CredentialCatalogItem>>([])
+  const [openID4VCIClient, setOpenID4VCIClient] = useState<OpenID4VCIClient>()
+
+  const { isLoading, isError, data: partyData } = useOne<Party, HttpError>({
+    resource: 'parties',
+    id
+  })
+
+  const getSupportedCredentials = (): Record<string, CredentialConfigurationSupportedV1_0_13> => { // TODO any
+    try {
+      // TODO returns a map not an array
+      //const supportedCredentials = openID4VCIClient!.getCredentialsSupported() // TODO !
+      const credentialsSupported: Record<string, CredentialConfigurationSupportedV1_0_13> = {
```

Comment:
Should get these from the issuer/contact using the oid4vciclient as commented out on the first line. I think we create an array of objects from the record. Which is fine

Reply:
fixed